server/src/main/java/org/elasticsearch/index/IndexSettings.java

```diff
@@ -303,11 +307,23 @@ public void validate(final TimeValue value, final Map<Setting<?>, Object> settin
             final Boolean fastRefresh = (Boolean) settings.get(INDEX_FAST_REFRESH_SETTING);
             final IndexVersion indexVersion = (IndexVersion) settings.get(SETTING_INDEX_VERSION_CREATED);
 
+            final boolean allowOverride = Boolean.parseBoolean(
+                System.getProperty(STATELESS_ALLOW_INDEX_REFRESH_INTERVAL_OVERRIDE, "false")
+            );
+
             if (existingShardsAllocator.equals("stateless")
                 && fastRefresh == false
                 && value.compareTo(TimeValue.ZERO) > 0
                 && value.compareTo(STATELESS_MIN_NON_FAST_REFRESH_INTERVAL) < 0
                 && indexVersion.after(IndexVersions.V_8_10_0)) {
+
+                if (allowOverride) {
+                    final Logger logger = LogManager.getLogger(RefreshIntervalValidator.class);
+                    logger.warn("Overriding `index.refresh_interval` setting to {}", value);
+                    // allow `index.refresh_interval` setting be in a range (0 .. `STATELESS_MIN_NON_FAST_REFRESH_INTERVAL`)
+                    return;
```

Comment:
Can we use an else statement instead? That would seem more intuitive here rather than explicit control flow.

Reply:
Removed as per comment below